apptools/preferences/preferences_helper.py

```diff
@@ -75,21 +75,20 @@ def _anytrait_changed(self, trait_name, old, new):
         if self.preferences is None:
             return
 
+        # If we were the one that set the trait (because the underlying
+        # preferences node changed) then do nothing.
+        if self._is_preference_trait(trait_name):
+            self.preferences.set("%s.%s" % (self._get_path(), trait_name), new)
```

Comment:
I do find this comment a little confusing... We are not conditionally doing nothing as it suggests

Reply:
Good point. I think I know what this means, but I agree this is not obvious at all, I will try to fix this comment.